brain/morphologies.cpp

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "morphologies.h"
+#include "morphology.h"
+
+#include <brion/blueConfig.h>
+#include <brion/circuit.h>
+#include <brion/morphologies.h>
+#include <brion/morphology.h>
+
+namespace brain
+{
+namespace morphologies
+{
+
+namespace
+{
+
+Morphologies _transformMorphologies( const brion::Morphologies& raw,
+                                     const brion::Circuit& circuit,
+                                     const brion::GIDSet& gids )
+{
+    const brion::NeuronMatrix& data = circuit.get(
+        gids, brion::NEURON_POSITION_X | brion::NEURON_POSITION_Y |
+             brion::NEURON_POSITION_Z | brion::NEURON_ROTATION );
+
+    std::vector< brion::Matrix4f > transforms(
+        gids.size(), brion::Matrix4f::IDENTITY );
+    #pragma omp parallel for
+    for( size_t i = 0; i < gids.size(); ++i )
+    {
+        brion::Matrix4f& matrix = transforms[i];
+        matrix.rotate(
+            boost::lexical_cast< float >( data[i][3] ) * ( M_PI/180.0f ),
+            brion::Vector3f( 0, 1, 0 ));
+        matrix.set_translation(
+            brion::Vector3f( boost::lexical_cast< float >( data[i][0] ),
+                             boost::lexical_cast< float >( data[i][1] ),
+                             boost::lexical_cast< float >( data[i][2] )));
+    }
+
+    Morphologies result;
+    for( size_t i = 0; i < raw.size(); ++i )
+    {
+        MorphologyPtr morphology( new Morphology( raw[i] ));
+        morphology->transform( transforms[i] );
+        morphology->setGlobal();
+        result.push_back( morphology );
+    }
+    return result;
+}
+
+}
+
+Morphologies load( const brion::BlueConfig& config, const brion::GIDSet& gids )
```

Comment:
this has been discussed I know but !on the API level having two function names similar doing different things is weird. It can be loadToGIDs and loadFromGIDs instead of being confusing ?